cli/types/cypress.d.ts

```diff
@@ -175,6 +175,13 @@ declare namespace Cypress {
    */
   type Config = ResolvedConfigOptions & RuntimeConfigOptions
 
+  type AddSubjectArgument<
+    TPrevSubject extends CommandOptions['prevSubject'],
+    TFunction extends (...args: any) => any
+    > = (
+    ...args: [TPrevSubject extends true ? JQuery | HTMLElement : undefined | JQuery | HTMLElement, ...Parameters<TFunction>]
```

Comment:
I appreciated looking at your fix in https://github.com/cypress-io/cypress/issues/18879! 

When trying to pull it into my own custom method code for which the recent TypeScript changes started raising errors, I realized that your solution would not work for me because the `Subject` some of my custom commands interact with are not necessarily underlying HTML subjects, e.g. `cy.getUser('someSeederIdentifier').loginAsUser()`, where `getUser` is a custom method that generates a test user from the command line, and `loginAsUser` takes in the previous subject, but it is just a blob of data rather than an element. 

I'm open to the idea that this is not an "appropriate" use of the Subject, but the Cypress docs only enforces the type of the Subject if you pass in one of the special keys (`'element', 'document', 'window'`). Otherwise Subject could technically be anything. https://docs.cypress.io/api/cypress-api/custom-commands#Usage

I wonder if we'd be able to update the generic usage in `add` to allow for more flexibility of the `subject` type?

Reply:
Thanks for the tip about `'element', 'document', 'window'` - I couldn't figure out how to get the generic to chain, but I did update the types to take into consideration the string values. My updated code has `true, 'optional'` setting the arg to "any" so we can set the value in the arguments of the function.

```tsx
Cypress.Commands.add(
  'command',
  { prevSubject: true },
  (prev: SomeType, actionKey) => // as 'prev' is any we can force it to what we know it is in the args. 
```